firestore-translate-text/functions/src/index.ts

```diff
@@ -179,25 +273,32 @@ const translateSingle = async (
 
 const translateMultiple = async (
   input: object,
-  snapshot: admin.firestore.DocumentSnapshot
+  snapshot: admin.firestore.DocumentSnapshot,
+  keepExistingTranslations: boolean = false
 ): Promise<void> => {
   let translations = {};
   let promises = [];
+  const existingTranslations = extractOutput(snapshot);
 
-  Object.entries(input).forEach(([input, value]) => {
-    config.languages.forEach((language) => {
+  Object.entries(input).forEach(([entry, value]) => {
+    const languages = config.languages.filter(
+      (targetLanguage: string): boolean =>
+        !keepExistingTranslations ||
```

Comment:
Same surprising if condition here

Reply:
Fixed to match the behavior of translateSingle (and include logging)